oval_graph/converter.py

```diff
@@ -32,6 +32,8 @@ def __init__(self, tree):
 
         if isinstance(tree, OvalNode):
             self.tree = tree
+            self.result = (self.tree.evaluate_tree()
+                           if self.tree.evaluate_tree() else self.tree.value)
```

Comment:
I think that unless there is some clever magic this code means that the whole tree twice.

What about something like:
```
self.result = self.result.evaluate_tree()
if not self.result:
    self.result = self.tree.value
```

Feel free to reply to the comments. The pull request review isn't supposed to be a one-way order trasmission but rather a balanced discussion :smile: